src/shelly_switch.cpp

```diff
@@ -316,4 +325,36 @@ void ShellySwitch::InputEventHandler(Input::Event ev, bool state) {
   }
 }
 
+void ShellySwitch::AddPowerMeter(uint16_t *iid) {
+  if (out_pm_ == nullptr) return;
+
+  // Power
+  power_char = new mgos::hap::UInt16Characteristic(
+      (*iid)++, &kHAPCharacteristic_EveConsumption, 0, 65535, 1,
+      [this](HAPAccessoryServerRef *,
+             const HAPUInt16CharacteristicReadRequest *, uint16_t *value) {
+        *value = out_pm_->GetPowerW().ValueOrDie();
+        return kHAPError_None;
+      },
+      true /* supports_notification */, nullptr, "eve-power-consumption");
+  AddChar(power_char);
+  // Energy
+  total_power_char = new mgos::hap::UInt16Characteristic(
+      (*iid)++, &kHAPCharacteristic_EveTotalConsumption, 0, 65535, 1,
+      [this](HAPAccessoryServerRef *,
+             const HAPUInt16CharacteristicReadRequest *, uint16_t *value) {
+        *value = out_pm_->GetEnergyWH().ValueOrDie() / 1000.0f;
+        return kHAPError_None;
+      },
+      true /* supports_notification */, nullptr, "eve-total-power-consumption");
+  AddChar(total_power_char);
+
+  power_timer_.Reset(1000, MGOS_TIMER_REPEAT);
```

Comment:
1 second seems too often, will cause a lot of traffic.
please change to, like 10 seconds and add some fuzz, e.g. +/- 500 ms (use [mgos_rand_range](mgos_rand_range)).

Reply:
I found already 1 sec long. What do you think about keeping 1 sec but only firing an event if value is changed?

Reply:
firing if consumption changes SGTM, thats' exactly how it's supposed to work.